File: app/src/main/java/com/github/ilikeyourhat/sudokusolver/solving/deduction/solver/DeductionSolverV1.java
package com.github.ilikeyourhat.sudokusolver.solving.deduction.solver;

import com.github.ilikeyourhat.sudokusolver.solving.deduction.algorithm.DeductionAlgorithm;
import com.github.ilikeyourhat.sudokusolver.solving.deduction.algorithm.HiddenSinglesAlgorithm;
import com.github.ilikeyourhat.sudokusolver.solving.deduction.algorithm.HintEliminationAlgorithm;
import com.github.ilikeyourhat.sudokusolver.solving.deduction.algorithm.NakedSinglesAlgorithm;
import com.github.ilikeyourhat.sudokusolver.model.type.SudokuType;

import java.util.ArrayList;
import java.util.List;

public class DeductionSolverV1 extends DeductionSolver {

    @Override
    protected List<DeductionAlgorithm.Factory> provideAlgorithms(SudokuType type) {
        List<DeductionAlgorithm.Factory> algorithms = new ArrayList<>();
        algorithms.add(new HintEliminationAlgorithm.Factory());
        algorithms.add(new NakedSinglesAlgorithm.Factory());
        algorithms.add(new HiddenSinglesAlgorithm.Factory());
        return algorithms;
    }
}

File: src/main/kotlin/io/github/ilikeyourhat/kudoku/solving/deduction/solver/DeductionSolverV1.kt
package io.github.ilikeyourhat.kudoku.solving.deduction.solver

import io.github.ilikeyourhat.kudoku.model.SudokuType
import io.github.ilikeyourhat.kudoku.solving.deduction.algorithm.DeductionAlgorithm
import io.github.ilikeyourhat.kudoku.solving.deduction.algorithm.HiddenSinglesAlgorithm
import io.github.ilikeyourhat.kudoku.solving.deduction.algorithm.HintEliminationAlgorithm
import io.github.ilikeyourhat.kudoku.solving.deduction.algorithm.NakedSinglesAlgorithm

class DeductionSolverV1 : DeductionSolver() {

    override fun provideAlgorithms(type: SudokuType): List<DeductionAlgorithm.Factory> {
        return listOf(
            HintEliminationAlgorithm.Factory(),
            NakedSinglesAlgorithm.Factory(),
            HiddenSinglesAlgorithm.Factory()
        )
    }
}
